DIRHASH_STANDARD.md

```diff
@@ -0,0 +1,203 @@
+
+
+
+
+## Protocol
+The `DIRHASH` of a directory is obtained by taking the underlying hash function's hexdigest of a `DIR-DESCRIPTOR` string. The `DIR-DESCRIPTOR` is composed by concatenation of an ordered sequence of `ENTRY-DESCRIPTOR`:s, each followed by a newline:
+
+```<entry-descriptor>\n<entry-descriptor>...<entry-descriptor>\n```,
+
+in python:
+
+```python
+dir_descriptor = '\n'.join(sorted(entry_descriptors) + [''])
+```
+
+A *directory entry* is either a subdirectory, a file or a symbolic link. Other file types (named pipe, socket, device file, door) are excluded in the core version of the Dirhash Standard. No distinction is made between files and "hard links" to files.
+
+The `ENTRY-DESCRIPTOR` is composed by concatenation of an ordered sequence of entry properties separated by the [null character](https://en.wikipedia.org/wiki/Null_character) `\000`. Each property is represented by its name and value separated by a colon `:`:
+
+```<name>:<value>\000<name>:<value>...\000<name>:<value>```,
+
+in python:
+
+```python
+entry_properties = {'property_1': 'value_1', ...}
+entry_property_strings = ['{}:{}'.format(k, v) for k, v in entry_properties.items()]
+entry_descriptor = '\000'.join(sorted(entry_property_strings))
+```
+The null character is the only character not allowed in property names or values (to maintain collision resistance).
+
+
+### Entry Properties
+An entry property refers to data or metadata of a directory entry.
+
+Name  | Value | Inclusion | Comment/Rationale
+----- | ----- | --------- | -----------------
+dirhash | The `DIRHASH` of a subdirectory or the target directory in case of a symbolic link, except for [cyclic links](#cyclic-links). | Always included for directories. *Not applicable to files*. | This is the recursive part of the Dirhash Protocol; the content of each subdirectory is summarized by its `DIRHASH`.
+data | Hash function hexdigest of the binary data of the file, or the file linked to, if a symlink. | Optional, but one of `name` and `data` must always be included. *Not applicable to directories*. | For the typical use case, the data should affect the hash. Without it, only the paths to files and subdirectories are hashed. 
+name | The name of the entry (the name of the link itself if a symlink, *not* the entry linked to). | Optional, but one of `name` and `data` must always be included. | For the typical use case, the entry name should affect the hash, so that data and other metadata is tied to the name and, subsequently, to the entry's relative path to the Dirhash root (which follows from the recursive nature of the Dirhash Standard).
+is_link | Whether the entry is a symlink; `true` or `false`. | Optional. | For the typical use case, it does *not* matter if a file or directory is linked or not - the file tree is "perceived the same" for many applications. If it matters, this property can be included. 
```

Comment:
clearify that it's refering to the string `true` `false` and not the value, (if that's the case)